src/main/java/appeng/fluids/parts/PartFluidImportBus.java

```diff
@@ -0,0 +1,201 @@
+		this.getConfigManager().registerSetting( Settings.REDSTONE_CONTROLLED, RedstoneMode.IGNORE );
+		this.getConfigManager().registerSetting( Settings.FUZZY_MODE, FuzzyMode.IGNORE_ALL );
+		this.getConfigManager().registerSetting( Settings.CRAFT_ONLY, YesNo.NO );
+		this.getConfigManager().registerSetting( Settings.SCHEDULING_MODE, SchedulingMode.DEFAULT );
+		this.source = new MachineSource( this );
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( IGridNode node )
+	{
+		return new TickingRequest( TickRates.FluidImportBus.getMin(), TickRates.FluidImportBus.getMax(), this.isSleeping(), false );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( IGridNode node, int ticksSinceLastCall )
+	{
+		return this.canDoBusWork() ? this.doBusWork() : TickRateModulation.IDLE;
+	}
+
+	@Override
+	protected boolean canDoBusWork()
+	{
+		return this.getProxy().isActive();
+	}
+
+	@Override
+	protected TickRateModulation doBusWork()
+	{
+		if( !this.canDoBusWork() )
+		{
+			return TickRateModulation.IDLE;
+		}
+
+		final TileEntity te = this.getConnectedTE();
+		if( te != null && te.hasCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() ) )
+		{
+			try
+			{
+				final IFluidHandler fh = te.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() );
+				final IMEMonitor<IAEFluidStack> inv = this.getProxy().getStorage().getInventory( this.getChannel() );
+				if( fh != null )
+				{
+					FluidStack fluidStack = fh.drain( this.calculateAmountToSend(), false );
+					if( this.filterEnabled() && !this.isInFilter( fluidStack ) )
+					{
+						return TickRateModulation.SLOWER;
+					}
+					AEFluidStack aeFluidStack = AEFluidStack.fromFluidStack( fluidStack );
+					if( aeFluidStack != null )
+					{
+						IAEFluidStack notInserted = inv.injectItems( aeFluidStack, Actionable.MODULATE, this.source );
+						if( notInserted != null && notInserted.getStackSize() > 0 )
+						{
+							aeFluidStack.decStackSize( notInserted.getStackSize() );
+						}
+						fh.drain( aeFluidStack.getFluidStack(), true );
+						return TickRateModulation.FASTER;
+					}
+				}
+			}
+			catch( GridAccessException e )
+			{
+				e.printStackTrace();
+			}
+		}
+
+		return TickRateModulation.SLEEP;
+	}
+
+	private boolean isInFilter( FluidStack fluid )
+	{
+		for( int i = 0; i < this.getConfig().getSlots(); i++ )
+		{
+			IAEItemStack stack = this.getConfig().getAEStackInSlot( i );
+			if( stack != null && stack.getDefinition() != null )
+			{
+				IFluidHandlerItem fh = stack.getDefinition().getCapability( CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null );
+				if( fh == null )
+				{
+					throw new NullPointerException( "IFluidHandlerItem is null" );
+				}
+				FluidStack filtered = fh.drain( Integer.MAX_VALUE, false );
+				if( filtered != null && filtered.isFluidEqual( fluid ) )
+				{
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private boolean filterEnabled(){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)